internal/controllers/clusterextension_controller_test.go

```diff
@@ -1205,6 +1201,85 @@ func TestSetDeprecationStatus(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "deprecated channels specified, ChannelDeprecated and Deprecated status set to true, others set to false",
+			clusterExtension: &ocv1alpha1.ClusterExtension{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Spec: ocv1alpha1.ClusterExtensionSpec{
+					Source: ocv1alpha1.SourceConfig{
+						SourceType: "Catalog",
+						Catalog: &ocv1alpha1.CatalogSource{
+							Channels: []string{"badchannel", "anotherbadchannel"},
+						},
+					},
+				},
+				Status: ocv1alpha1.ClusterExtensionStatus{
+					Conditions: []metav1.Condition{},
+				},
+			},
+			expectedClusterExtension: &ocv1alpha1.ClusterExtension{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Spec: ocv1alpha1.ClusterExtensionSpec{
+					Source: ocv1alpha1.SourceConfig{
+						SourceType: "Catalog",
+						Catalog: &ocv1alpha1.CatalogSource{
+							Channels: []string{"badchannel", "anotherbadchannel"},
+						},
+					},
+				},
+				Status: ocv1alpha1.ClusterExtensionStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:               ocv1alpha1.TypeDeprecated,
+							Reason:             ocv1alpha1.ReasonDeprecated,
+							Status:             metav1.ConditionTrue,
+							ObservedGeneration: 1,
+						},
+						{
+							Type:               ocv1alpha1.TypePackageDeprecated,
+							Reason:             ocv1alpha1.ReasonDeprecated,
+							Status:             metav1.ConditionFalse,
+							ObservedGeneration: 1,
+						},
+						{
+							Type:               ocv1alpha1.TypeChannelDeprecated,
+							Reason:             ocv1alpha1.ReasonDeprecated,
+							Status:             metav1.ConditionTrue,
+							ObservedGeneration: 1,
```

Comment:
Hmm, I didn't realize this test ignored the message. Follow-up?

Reply:
Is there a specific reason we should be testing the message?

I understand wanting to verify our message crafting logic is correct, but I lean towards the camp where the message is any arbitrary string and is not part of our API guarantees - testing for it explicitly makes our unit tests brittle